### DISCUSSION TITLE
How to attach the debugger to a local windows process?

### DISCUSSION THREAD

**Discussion by matcool** [NationalSecurityAgency/ghidra#3092] [Debugger]
I have figured out how to launch the current program, but ends up not working, so my only alternative is to attach to an already running process, but i haven't been able to figure that out

**Comment by d-millar** [comment]
@matcool yes, not entirely obvious I will concede, but two ways to do it:

(1) In the Objects viewer, open up the "Available" node.  On a good day, your process's pid will be in the list.  If not, use the refresh button to make it so.  (Apologies for having just pids there - names would help, I know.)  Possibly useful in this context is the 'b' button, which will change the base for the pids.  Once you've found it, select it and hit the "Attach" button.

(2) Ignore "Available" and get the pid by whatever means you find easiest.  Open up the "Connectors" node in Objects and select the "Attach" version.  Then hit the "Launch" button (not "Attach - I know, I know) and enter the pid in the parameters field.

Yell if these don't work!
